src/atl/dependencygraph.rs

```diff
@@ -0,0 +1,501 @@
+                    if props.contains(&prop) {
+                        let mut edges: HashSet<Edges<ATLVertex>> = HashSet::new();
+                        edges.insert(Edges::HYPER(HyperEdge {
+                            source: vert.clone(),
+                            targets: vec![],
+                        }));
+                        edges
+                    } else {
+                        HashSet::new()
+                    }
+                }
+                Phi::NOT(phi) => {
+                    let mut edges: HashSet<Edges<ATLVertex>> = HashSet::new();
+
+                    edges.insert(Edges::NEGATION(NegationEdge {
+                        source: vert.clone(),
+                        target: ATLVertex::FULL {
+                            state: *state,
+                            formula: phi.clone(),
+                        },
+                    }));
+
+                    edges
+                }
+                Phi::AND(left, right) => {
+                    let mut edges = HashSet::new();
+
+                    let left_targets = vec![ATLVertex::FULL {
+                        state: *state,
+                        formula: left.clone(),
+                    }];
+                    edges.insert(Edges::HYPER(HyperEdge {
+                        source: vert.clone(),
+                        targets: left_targets,
+                    }));
+
+                    let right_targets = vec![ATLVertex::FULL {
+                        state: *state,
+                        formula: right.clone(),
+                    }];
+                    edges.insert(Edges::HYPER(HyperEdge {
+                        source: vert.clone(),
+                        targets: right_targets,
+                    }));
+
+                    edges
+                }
+                Phi::NEXT { players, formula } => {
+                    let moves: Vec<usize> = self
+                        .game_structure
+                        .move_count(*state)
+                        .iter()
+                        .map(|&count| count as usize)
+                        .collect();
+                    VarsIterator::new(moves, players.iter().copied().collect())
+                        .map(|pmove| {
+                            let targets: Vec<ATLVertex> =
+                                DeltaIterator::new(&self.game_structure, *state, pmove)
+                                    .map(|state| ATLVertex::FULL {
+                                        state,
+                                        formula: formula.clone(),
+                                    })
+                                    .collect();
+                            Edges::HYPER(HyperEdge {
+                                source: vert.clone(),
+                                targets,
+                            })
+                        })
+                        .collect::<HashSet<Edges<ATLVertex>>>()
+                }
+                Phi::UNTIL {
```

Comment:
Så vidt jeg kan se, har du prøvet at lave despite-until (Figur 6), men du har ikke fået inverteret spillerne. Bemærk at det er de *andre* spillere, der bruges i `vars` i sidebetængelsen fra Figur 6.

Reply:
fixed in 0a41d11f149a2df0c0a4a134b87262e940282353